### PR TITLE
Allow MSQ to have an "other" field where users can specify their own answer #1740

### LINKED ISSUE #1740: Allow MSQ to have an "other" field where users can specify their own answer
_From [arnold.k...@gmail.com](https://code.google.com/u/105779801956403207546/) on March 21, 2014 13:47:53_

Similar with issue 1423 but for MSQ

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1743_

### PR DESCRIPTION
**Pull request by taniach**
Fixes #1740

### CODE REVIEW

**Comment by JosephineKwa** [comment]
I think it would be a good idea to apply bootstrap to the "other" input text box too (to the single-answer one as well), to keep the design consistent. You could open a new issue to regarding this.

File: src/main/java/teammates/common/datatransfer/FeedbackMsqQuestionDetails.java [hunk 1]
```diff
@@ -350,20 +408,47 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
...
-            List<String> answerStrings = ((FeedbackMsqResponseDetails)response.getResponseDetails()).getAnswerStrings();
+            List<String> answerStrings = ((FeedbackMsqResponseDetails)response.getResponseDetails()).getAnswerStrings(); 
+            Boolean isOtherOptionAnswer = ((FeedbackMsqResponseDetails) (response.getResponseDetails())).isOtherOptionAnswer();
+            String otherAnswer = "";
+            
+            if (isOtherOptionAnswer) {
+                if (!answerFrequency.containsKey("Other")) {
+                    answerFrequency.put("Other", 1);
+                } else {
+                    answerFrequency.put("Other", answerFrequency.get("Other")+1);
```
Comment (JosephineKwa) [inline src/main/java/teammates/common/datatransfer/FeedbackMsqQuestionDetails.java]: Remember to add spaces between the `+`, similarly for the one on line 445, 508 and 526 :)

File: src/main/java/teammates/common/datatransfer/FeedbackMsqQuestionDetails.java [hunk 2]
```diff
@@ -42,7 +43,12 @@ public boolean extractQuestionDetails(
             FeedbackQuestionType questionType) {
         int numOfMsqChoices = 0;
         List<String> msqChoices = new LinkedList<String>();
-        boolean msqOtherEnabled = false; // TODO change this when implementing "other, please specify" field
+        boolean msqOtherEnabled = false; 
+        
+        if (HttpRequestHelper.getValueFromParamMap(requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MSQOTHEROPTIONFLAG) != null
```
Comment (JosephineKwa) [inline src/main/java/teammates/common/datatransfer/FeedbackMsqQuestionDetails.java]: You could perhaps extract this into a local variable for clarity

**Comment by JosephineKwa** [comment]
Some changes requested

**Comment by taniach** [comment]
Created #4014 for using Bootstrap for other option textbox

**Comment by taniach** [comment]
@JosephineKwa ready for review again

**Comment by JosephineKwa** [comment]
Good for merging once dev green

File: src/main/webapp/js/feedbackSubmissionsEdit.js [hunk 3]
```diff
@@ -158,7 +174,7 @@ function prepareContribQuestions() {
         var qnNum = contribQuestionNums[i];
 
         // Get number of options for the specified question number of contribution question type
-        var optionNums = $('[name^="responsetext-' + qnNum + '-"]').length;
+        var optionNums = $('[name^="responsetext-' + qnNum + '"]').length;
```
Comment (kanghj) [inline src/main/webapp/js/feedbackSubmissionsEdit.js]: @tania, I think you accidentally reverted some changes in an earlier pull request here?

**Pull request merged**